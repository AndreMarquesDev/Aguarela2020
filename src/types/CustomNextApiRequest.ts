import type { NextApiRequest } from 'next';
import type { FormPostRequestBody } from './FormPostRequestBody';

export type CustomNextApiRequest = {
    body: FormPostRequestBody;
} & NextApiRequest;
