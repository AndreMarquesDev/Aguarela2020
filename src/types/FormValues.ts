export type FormValues = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};
