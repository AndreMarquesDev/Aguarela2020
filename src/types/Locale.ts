export enum Locale {
    Pt = 'pt',
    En = 'en',
}
