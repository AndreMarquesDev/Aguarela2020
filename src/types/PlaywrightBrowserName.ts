export enum PlaywrightBrowserName {
    Chromium = 'chromium',
    Firefox = 'firefox',
    Webkit = 'webkit',
}
