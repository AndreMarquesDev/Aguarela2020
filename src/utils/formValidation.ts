export enum FieldTypes {
    Name = 'name',
    Email = 'email',
    Textarea = 'message',
    Brand = 'brand',
    Subject = 'subject',
}
export type FormState = {
    name: string;
    email: string;
    message: string;
    brand: string;
    subject: string;
};

export const validateName = (value: string): boolean => {
    const nameRegex =
        // eslint-disable-next-line regexp/no-unused-capturing-group, regexp/prefer-plus-quantifier, regexp/prefer-character-class, regexp/no-super-linear-backtracking, regexp/no-useless-escape
        /s*([a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]{1,}([\.,] |[-']| ))+[a-zA-ZáâàãéêèíîìóôòõúûùñçäöüßÁÂÀÃÉÊÈÍÎÌÓÔÒÕÚÛÙÑÇÄÖÜẞ]+\.?\s*$/;

    return nameRegex.test(value);
};

export const validateEmail = (value: string): boolean => {
    const emailRegex =
        // eslint-disable-next-line regexp/no-useless-escape, regexp/no-unused-capturing-group, regexp/prefer-d, regexp/use-ignore-case
        /(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

    return emailRegex.test(value);
};
