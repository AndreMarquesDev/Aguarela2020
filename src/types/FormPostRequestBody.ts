import { FormValues } from './FormValues';

export interface FormPostRequestBody extends FormValues {
    isPlaywright: boolean;
}
