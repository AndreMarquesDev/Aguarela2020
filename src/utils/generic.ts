import { Locale } from '../types/Locale';
import { textsPt, textsEn, TextsInterface } from './texts';

export const isDev = process.env.NODE_ENV === 'development';
export const capitalize = (string: string): string =>
    string.charAt(0).toUpperCase() + string.slice(1);
export const getCurrentLanguagetexts = (language: Locale): TextsInterface =>
    language === Locale.Pt ? textsPt : textsEn;

export const resetTimesTouchedAttribute = (): void =>
    document.querySelectorAll('[data-times-touched]').forEach(
        element => ((element as HTMLLIElement).dataset.timesTouched = '0') // eslint-disable-line no-return-assign, no-param-reassign
    );
