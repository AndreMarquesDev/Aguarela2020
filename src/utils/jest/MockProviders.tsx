/* eslint-disable react/jsx-no-constructed-context-values */
import React, { ReactNode } from 'react';
import { NavLinksContext } from '../../components/context/NavLinksContext';
import { TextsContext } from '../../components/context/TextsContext';
import { Locale } from '../../types/Locale';
import { textsEn, textsPt } from '../texts';

interface MockProvidersProps {
    language?: Locale;
    isMenuOpen?: boolean;
    toggleMenu?: jest.Mock;
    setNavHeight?: jest.Mock;
    children: ReactNode;
}

export const MockProviders = ({
    language = Locale.Pt,
    isMenuOpen = false,
    toggleMenu = jest.fn(),
    setNavHeight = undefined,
    children,
}: MockProvidersProps): JSX.Element => {
    return (
        <TextsContext.Provider
            value={{
                texts: language === Locale.Pt ? textsPt : textsEn,
            }}
        >
            <NavLinksContext.Provider value={{ isMenuOpen, toggleMenu, setNavHeight }}>
                {children}
            </NavLinksContext.Provider>
        </TextsContext.Provider>
    );
};
