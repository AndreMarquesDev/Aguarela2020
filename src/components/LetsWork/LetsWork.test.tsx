import type { RenderResult } from '@testing-library/react';
import { render, screen } from '@testing-library/react';
import React from 'react';
import { Locale } from '../../types/Locale';
import { MockProviders } from '../../utils/jest/MockProviders';
import { textsEn, textsPt } from '../../utils/texts';
import { LetsWork } from './LetsWork';
import '@testing-library/jest-dom';

const renderComponent = (language: Locale = Locale.Pt): RenderResult => {
    return render(
        <MockProviders language={language}>
            <LetsWork />
        </MockProviders>,
    );
};

describe('<LetsWork />', () => {
    test('renders properly', () => {
        const { container } = renderComponent();

        expect(screen.getByText(textsPt.letsWork)).toBeInTheDocument();
        expect(screen.getByText(textsPt.letsWorkDescription)).toBeInTheDocument();
        expect(screen.getByText(textsPt.contact)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });

    test('renders properly in English', () => {
        const { container } = renderComponent(Locale.En);

        expect(screen.getByText(textsEn.letsWork)).toBeInTheDocument();
        expect(screen.getByText(textsEn.letsWorkDescription)).toBeInTheDocument();
        expect(screen.getByText(textsEn.contact)).toBeInTheDocument();

        expect(container).toMatchSnapshot();
    });
});
