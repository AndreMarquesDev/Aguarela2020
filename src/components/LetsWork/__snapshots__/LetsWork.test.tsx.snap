// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LetsWork /> renders properly 1`] = `
<div>
  <section
    class="jsx-1811239983 container"
    data-testid="letsWork_section"
  >
    <div
      class="jsx-1811239983 wrapper genericMargins"
    >
      <h1
        class="jsx-3953086084 "
      >
        Vamos trabalhar juntos!
      </h1>
      <p
        class="jsx-1811239983 bodyText bodyText--white"
      >
        Precisa de ajuda com as redes sociais e marketing digital da sua marca ou negócio? Envie-me um email e vamos conversar. Estou disponível para ajudar a posicionar a sua marca no mundo digital.
      </p>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <a
            class="jsx-2566809609 button"
            href="/pt/contact"
          >
            Contacto
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`<LetsWork /> renders properly in English 1`] = `
<div>
  <section
    class="jsx-1811239983 container"
    data-testid="letsWork_section"
  >
    <div
      class="jsx-1811239983 wrapper genericMargins"
    >
      <h1
        class="jsx-3953086084 "
      >
        Let's work together!
      </h1>
      <p
        class="jsx-1811239983 bodyText bodyText--white"
      >
        Need help with digital marketing and the social media of your brand or business? Send me an email and let's talk. I'm available to help position your brand in the digital world.
      </p>
      <div
        class="jsx-2566809609 buttonWrapper"
      >
        <div
          class="jsx-2566809609 buttonBackground alignCenter"
        >
          <a
            class="jsx-2566809609 button"
            href="/pt/contact"
          >
            Contact
          </a>
        </div>
      </div>
    </div>
  </section>
</div>
`;
